### PR TITLE
Added information about deployer 2.0

### PR DESCRIPTION
**Pull request by meker12**
Integrate updates from #8005 

## Purpose of this pull request

This pull request (PR) adds information about the new features on the command to deploy instances for issues and PRs.

## Affected DevDocs pages

-  https://devdocs.magento.com/contributor-guide/contributing.html

### CODE REVIEW

File: src/contributor-guide/contributing.md [hunk 1]
```diff
@@ -248,6 +248,78 @@ Admins access will be shared via comment on GitHub.
 -  Magento EngCom Team
 -  [Contributor]({{ site.baseurl }}/contributor-guide/contributors.html)
 
+### Customize deployed instances {#customize-deployed-instances}
+
+In some cases a custom environment is required to test an issue or a pull request. You can create a custom environment by appending custom configuration settings to the PR comment to [Deploy a vanilla Magento instance](#vanilla-pr) or [Deploy an instance based on PR changes](#deploy-pr).
+
+#### Magento Edition:
```
Comment (meker12) [inline src/contributor-guide/contributing.md:255]:
```suggestion
#### Magento edition
```

File: src/contributor-guide/contributing.md [hunk 2]
```diff
@@ -248,6 +248,78 @@ Admins access will be shared via comment on GitHub.
...
+```text
+with edition {$edition}
+```
+
+For`edition`, use either of the following values:
+
+-  `ee` deploys the Magento Commerce edition
+-  `b2b` deploys Magento Commerce with B2B modules.
+
+#### Environment:
```
Comment (meker12) [inline src/contributor-guide/contributing.md:268]:
```suggestion
#### Environment
```

File: src/contributor-guide/contributing.md [hunk 3]
```diff
@@ -248,6 +248,78 @@ Admins access will be shared via comment on GitHub.
...
+For example, append the following text to the PR comment to deploy an instance with PHP 7.4, Elasticsearch version 7, and MariaDB version 10.4.
+
+```text
+with env PHP 7.4, search-engine Elasticsearch 7, database MariaDB 10.4
+```
+
+{:.bs-callout-info}
+We generally recommend that you deploy the default environment. Use the custom configuration options only when you require a special configuration to test specfic use cases.
+
+#### Add Extensions:
```
Comment (meker12) [inline src/contributor-guide/contributing.md:295]:
```suggestion
#### Add extensions
```

File: src/contributor-guide/contributing.md [hunk 4]
```diff
@@ -248,6 +248,78 @@ Admins access will be shared via comment on GitHub.
...
+with extensions {$extensionRepo}
+```
+
+For `extensionRepo`, specify one or more extension repositories to include when compiling your instance.  If you specify multiple repositories, use a comma after each repository. You can specify a specific branch in a repository using the pattern: `org/repo-name:branch-name`.  For example:
+
+```text
+with extensions magento/security-package:1.0-develop, magento/security-package-ee
+```
+
+#### Remove Extensions:
```
Comment (meker12) [inline src/contributor-guide/contributing.md:309]:
```suggestion
#### Remove extensions
```

**Review by dobooth — APPROVED** [review]

**Comment by meker12** [comment]
running tests

**Pull request merged**